### PR TITLE
initial integration of the reading action

### PR DESCRIPTION
**Pull request by Arboghast**
Features include book selection, synchronous text audio and text highlighting, and automatic book progression upon successfully reading the text, or once the assistant is done giving feedback. Also a minimally styled Object Oriented frontend.

### CODE REVIEW

File: public/css/main.css [hunk 1]
```diff
@@ -1,15 +1,22 @@
-html,
-body {
+html {
```
Comment (gaurnett) [inline public/css/main.css:1]: All of these CSS changes will affect the overall game. We will have to review these changes together to see if this is best to make these changes.

Comment (gaurnett) [inline public/css/main.css:1]: All of these CSS changes will affect the overall game. We will have to review these changes together to see if this is best to make these changes.

File: public/css/reading.css [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    margin: 10vh 0;
+    padding: 0 10vw;
+    font-size: 40px;
+    text-align: center;
+}
+
+/* idk */
+#reading{
+    background-color: #f0d4d4;
+}
```
Comment (gaurnett) [inline public/css/reading.css]: nit: add new line

File: public/js/reading/ReadingMain.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+   */
+
+  view = document.createElement("div");
+  library = new Library();
+  text = new Text();
+
+  constructor() {
+    // set up fps monitoring
+    // const stats = new Stats();
+    // this.view.getElementsByClassName('stats')[0].appendChild(stats.domElement);
```
Comment (mithasgoogle) [inline public/js/reading/ReadingMain.js]: clean up unused code

File: sdk/webhooks/ActionsOnGoogleFulfillment/reading/mockTextData.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (mithasgoogle) [inline sdk/webhooks/ActionsOnGoogleFulfillment/reading/mockTextData.json]: Is this just for testing purposes right now, or will all the books be locally stored as a json?

**Review by mithasgoogle — APPROVED** [review]
Nice work, some comments inline.

**Comment by gaurnett** [comment]
Let's speak about the CSS changes you made before we merge.

File: public/js/main.js [hunk 5]
```diff
@@ -1,5 +1,5 @@
-import { Action } from "./action.js";
-import { Scene } from "./scene.js";
+import { Action } from "./Action.js";
```
Comment (gaurnett) [inline public/js/main.js:1]: Why did you change these to uppercase?

File: public/js/reading/ReadingMain.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import {Library} from './components/Library.js';
+import {Text} from './components/Text.js';
+/**
+ * Represent Home scene
+ */
+
+export class ReadingMain{
```
Comment (gaurnett) [inline public/js/reading/ReadingMain.js:7]: nit: add space between class name and curly bracket `export class ReadingMain {`

File: public/js/reading/ReadingMain.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  /**
+   * Initializes the game with visual components.
+   */
+
+  view = document.createElement("div");
+  library = new Library();
+  text = new Text();
+
+  constructor() {
+    // set up fps monitoring
```
Comment (gaurnett) [inline public/js/reading/ReadingMain.js]: nit: remove these comments

File: public/js/reading/components/Library.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+    libraryContainer = document.createElement("div"); //aka bookshelf
+
+    constructor() {
+        this.libraryContainer.classList.add("flex-center");
+        this.libraryContainer.setAttribute("id", "libraryContainer");
+        let books = document.createElement("div");
+        books.classList.add("bookshelf");
+        
+        //->temp place holder, make it dynamic
```
Comment (gaurnett) [inline public/js/reading/components/Library.js]: Fix comment here

File: sdk/custom/intents/READ_GOTO_LIBRARY.yaml [hunk 9]
```diff
@@ -0,0 +1,11 @@
+trainingPhrases:
```
Comment (gaurnett) [inline sdk/custom/intents/READ_GOTO_LIBRARY.yaml:1]: nit: change the file name to lowercase so that it is uniform with the rest of the project

File: sdk/custom/types/R_Book.yaml [hunk 10]
```diff
@@ -0,0 +1 @@
+freeText: {}
```
Comment (gaurnett) [inline sdk/custom/types/R_Book.yaml]: nit: change the file name to read_book.yaml

File: sdk/settings/settings.yaml [hunk 11]
```diff
@@ -1,8 +1,8 @@
 category: GAMES_AND_TRIVIA
 defaultLocale: en
 localizedSettings:
-    developerEmail:
-    developerName:
+    developerEmail: AOG2@ElectricBoogaloo.aol
```
Comment (gaurnett) [inline sdk/settings/settings.yaml:4]: What is this email?

**Review by gaurnett — COMMENTED** [review]
Overall good. A few comments
1. nit: I added a few nit comments that you should look into. There were repeats on other files but I didn't leave the same comments on each one.

File: sdk/webhooks/ActionsOnGoogleFulfillment/reading/mockTextData.json (hunk 4 above)
Comment (Arboghast) [reply sdk/webhooks/ActionsOnGoogleFulfillment/reading/mockTextData.json]: I plan on storing all the books locally, originally I was going to host it on firebase's realtime database but that option didn't seem useful vs storing locally.
